plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/BaseExplorerContentProvider.java

```diff
@@ -179,13 +179,37 @@ protected IStatus run(IProgressMonitor monitor) {
 		job.schedule();
 		return stub;
 	}
-
-	protected void refreshViewer(final Object object) {
+	
+	private void asyncViewerRefresh(final Runnable runner) {
```

Comment:
it would make it easier to understand if this method was called something like #asyncUpdateViewer